### PR TITLE
[5.2] have withoutOverlapping() method use the cache for storage of the lock file

### PR DESCRIPTION
**Pull request by tomschlick**
Useful for multiple server setups. Allows resiliency & the ability to store the lock file in any cache driver (file, database, redis, memcached, etc).

### CODE REVIEW

File: src/Illuminate/Console/Scheduling/Event.php [hunk 1]
```diff
@@ -64,6 +64,13 @@ class Event
     public $withoutOverlapping = false;
 
     /**
+     * Cache object used for $withoutOverlapping lock file.
+     * 
+     * @var Illuminate\Contracts\Cache\Repository
+     */
+    public $cache = null;
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: no need to have `= null`

File: src/Illuminate/Console/Scheduling/Event.php [hunk 2]
```diff
@@ -122,6 +129,7 @@ public function __construct($command)
     {
         $this->command = $command;
         $this->output = $this->getDefaultOutput();
+        $this->cache = new Illuminate\Contracts\Cache\Repository;
```
Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: this needs importing, and also wouldn't even work as it is now anyway

Comment (GrahamCampbell) [inline src/Illuminate/Console/Scheduling/Event.php]: please add a test

Comment (tomschlick) [inline src/Illuminate/Console/Scheduling/Event.php]: You're right... brain fart. I'll fix that and add tests.

**Comment by GrahamCampbell** [comment]
Please could you send to the new 5.2 branch. ;)

**Comment by GrahamCampbell** [comment]
Also, please only send one commit.

**Pull request closed without merging**